gctscript/modules/gct/common.go

```diff
@@ -0,0 +1,385 @@
+		{
+			indicators.MoneyFlowIndex,
+		},
+		{
+			"",
+		},
+	}
+
+	obj, ok := objects.ToInterface(a).(*indicators.MFI)
+	if !ok {
+		return nil, errors.New("casting failure")
+	}
+
+	for i := range obj.Value {
+		val, ok := objects.ToString(obj.Value[i])
+		if !ok {
+			return nil, errors.New("cannot convert object to string")
+		}
+		bucket = append(bucket, []string{val})
+	}
+	return bucket, nil
+}
+
+func convertOBV(a objects.Object) ([][]string, error) {
+	var bucket = [][]string{
+		{
+			indicators.OnBalanceVolume,
+		},
+		{
+			"",
+		},
+	}
+
+	obj, ok := objects.ToInterface(a).(*indicators.OBV)
+	if !ok {
+		return nil, errors.New("casting failure")
+	}
+
+	for i := range obj.Value {
+		val, ok := objects.ToString(obj.Value[i])
+		if !ok {
+			return nil, errors.New("cannot convert object to string")
+		}
+		bucket = append(bucket, []string{val})
+	}
+	return bucket, nil
+}
+
+func convertRSI(a objects.Object) ([][]string, error) {
+	var bucket = [][]string{
+		{
+			indicators.RelativeStrengthIndex,
+		},
+		{
+			"",
+		},
+	}
+
+	obj, ok := objects.ToInterface(a).(*indicators.RSI)
+	if !ok {
+		return nil, errors.New("casting failure")
+	}
+
+	for i := range obj.Value {
+		val, ok := objects.ToString(obj.Value[i])
+		if !ok {
+			return nil, errors.New("cannot convert object to string")
+		}
+		bucket = append(bucket, []string{val})
+	}
+	return bucket, nil
+}
+
+func convertSMA(a objects.Object) ([][]string, error) {
+	var bucket = [][]string{
+		{
+			indicators.SimpleMovingAverage,
+		},
+		{
+			"",
+		},
+	}
+
+	obj, ok := objects.ToInterface(a).(*indicators.SMA)
+	if !ok {
+		return nil, errors.New("casting failure")
+	}
+
+	for i := range obj.Value {
+		val, ok := objects.ToString(obj.Value[i])
+		if !ok {
+			return nil, errors.New("cannot convert object to string")
+		}
+		bucket = append(bucket, []string{val})
+	}
+	return bucket, nil
+}
+
+func convertOHLCV(a objects.Object) ([][]string, error) {
+	obj, ok := objects.ToInterface(a).(*OHLCV)
+	if !ok {
+		return nil, errors.New("casting failure")
+	}
+
+	exchange, ok := objects.ToString(obj.Value["exchange"])
+	if !ok {
+		return nil, errors.New("cannot convert object to string")
+	}
+
+	pair, ok := objects.ToString(obj.Value["pair"])
+	if !ok {
+		return nil, errors.New("cannot convert object to string")
+	}
+
+	asset, ok := objects.ToString(obj.Value["asset"])
+	if !ok {
+		return nil, errors.New("cannot convert object to string")
+	}
+
+	interval, ok := objects.ToString(obj.Value["intervals"])
+	if !ok {
+		return nil, errors.New("cannot convert object to string")
+	}
+
+	var bucket = [][]string{
+		{
+			indicators.OHLCV, "Exchange:" + exchange, pair, asset, interval, "",
+		},
+		{
+			"Date", "Open", "High", "Low", "Close", "Volume",
+		},
+	}
+
+	candles, ok := obj.Value["candles"]
+	if !ok {
+		return nil, errors.New("candles not found in object map")
+	}
+
+	data := candles.Iterate()
+
+	for data.Next() {
+		var date, open, high, low, close, volume string
```

Comment:
while Go doesn't restrict it "close" is a a built in name and its use for variables might cause confusion if close is ever used later